src/Kmd.Logic.Gateway.Automation/PublishPolicies.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Kmd.Logic.Gateway.Automation.Client;
+using Kmd.Logic.Gateway.Automation.Client.Models;
+using Kmd.Logic.Gateway.Automation.PublishFile;
+using Kmd.Logic.Identity.Authorization;
+
+namespace Kmd.Logic.Gateway.Automation
+{
+    internal class PublishPolicies
+    {
+        private readonly GatewayOptions options;
+        private readonly GatewayClientFactory gatewayClientFactory;
+        private List<GatewayAutomationResult> publishResults;
+
+        public PublishPolicies(HttpClient httpClient, LogicTokenProviderFactory tokenProviderFactory, GatewayOptions options, List<GatewayAutomationResult> publishResults)
+        {
+            this.options = options ?? throw new ArgumentNullException(nameof(options));
+            if (tokenProviderFactory == null)
+            {
+                throw new ArgumentNullException(nameof(tokenProviderFactory));
```

Comment:
if these validation are done in calling method, do we required here ?

Reply:
I had to check to avoid warning.